rosbag2_cpp/src/rosbag2_cpp/reindexer.cpp

```diff
@@ -99,33 +97,34 @@ bool Reindexer::compare_relative_file(
  *    The files will be `emplace_back`-ed on the passed vector
  */
 void Reindexer::get_bag_files(
-  const rcpputils::fs::path & base_folder,
-  std::vector<rcpputils::fs::path> & output)
+  const std::filesystem::path & base_folder,
+  std::vector<std::filesystem::path> & output)
 {
   std::regex regex_rule(regex_bag_pattern_, std::regex_constants::ECMAScript);
-  auto allocator = rcutils_get_default_allocator();
-  auto dir_iter = rcutils_dir_iter_start(base_folder.string().c_str(), allocator);
+  auto dir_iter = std::filesystem::directory_iterator(base_folder);
+  auto first = std::filesystem::begin(dir_iter);
+  auto last = std::filesystem::end(dir_iter);
 
   // Make sure there are files in the directory
-  if (dir_iter == nullptr) {
+  if (std::filesystem::is_empty(base_folder)) {
     throw std::runtime_error("Empty directory.");
   }
 
   // Get all file names in directory
-  do {
-    auto found_file = rcpputils::fs::path(dir_iter->entry_name);
-    ROSBAG2_CPP_LOG_DEBUG_STREAM("Found file: " << found_file.string());
+  for (; first != last; ++first) {
+    auto found_file = first->path().filename();
+    ROSBAG2_CPP_LOG_DEBUG_STREAM("Found file: " << found_file.generic_string());
```

Comment:
how about the following? so that we can delete the above local variables in stack?

```suggestion
  for (const auto& entry; std::filesystem::directory_iterator(base_folder)) {
    auto found_file = entry.path().filename();
    ROSBAG2_CPP_LOG_DEBUG_STREAM("Found file: " << found_file.generic_string());
```

Reply:
done, thanks